### PR TITLE
bundle analysis: introduce asset ordering in query

### PR DESCRIPTION
**Pull request by JerrySentry**
- Has an option to get list of assets ordered by name or size columns.
- Add `id` attribute to asset report, this will be useful for implementing pagination on the api side.

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: shared/bundle_analysis/report.py [hunk 1]
```diff
@@ -129,8 +133,11 @@ def asset_reports(
         asset_types: Optional[List[AssetType]] = None,
         chunk_entry: Optional[bool] = None,
         chunk_initial: Optional[bool] = None,
+        ordering_column: Optional[str] = "size",
```
Comment (JerrySentry) [inline shared/bundle_analysis/report.py:136]: Leaving it to callers that use `BundleReport` to validate existence of columns

File: shared/bundle_analysis/report.py [hunk 2]
```diff
@@ -142,7 +149,7 @@ def asset_reports(
                 asset_types,
                 chunk_entry,
                 chunk_initial,
-            )
+            ).order_by(ordering(getattr(Asset, ordering_column)))
```
Comment (giovanni-guidini) [inline shared/bundle_analysis/report.py:152]: Can you explain the purpose of this `getattr` there?... You are grabbing the value from the column listed in `ordering_column`. Is that the same as a reference to the column?

**Review by giovanni-guidini — APPROVED** [review]
LGTM but I didn't understand some implementation details.

File: shared/bundle_analysis/report.py (hunk 2 above)
Comment (JerrySentry) [reply shared/bundle_analysis/report.py:152]: It's doing something like `order_by(getattr(Asset, "size"))`, which is really just doing a `order_by(Asset.size)`.

**Pull request merged**